src/aws-cpp-sdk-core/include/smithy/identity/auth/AuthSchemeOption.h

```diff
@@ -19,9 +19,11 @@ namespace smithy {
         /* note: AuthSchemeOption is not connected with AuthScheme by type system, only by the String of schemeId, this is in accordance with SRA */
     public:
         AuthSchemeOption(const char* id = nullptr): schemeId(id) {}
+        AuthSchemeOption(const char* id, bool isStreaming): schemeId(id),isEventStreaming{isStreaming} {}
```

Comment:
please don't extended base interfaces with implementation details.
Just create a AuthScheme SigV4 EventStreaming.

Reply:
The way current code is designed otherwise needs way more changes. This minimizes that with default args